curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64cb522509ffb274daf9fd9e.md

```diff
@@ -0,0 +1,418 @@
+---
+id: 64cb522509ffb274daf9fd9e
+title: Step 105
+challengeType: 0
+dashedName: step-105
+---
+
+# --description--
+
+Create a new `const` variable called `checkpointDetectionRules` and assign it an empty array. 
+
+Inside that array, add a conditional statement that checks if the player's `position.x` is greater than or equal to the checkpoint's `position.x`.
+
+# --hints--
+
+TODO: Tried to add tests using `isArray` but it wasn't working. Need to come back to this.
+
+```js
+
+```
```

Comment:
These work:

You should use `const` to create an empty `checkpointDetectionRules` array.

```js
assert.match(code, /const\s+checkpointDetectionRules\s*=\s*\[\s*/)
```

You should have a conditional statement that checks if the player's `position.x` is greater than or equal to the checkpoint's `position.x` inside the `checkpointDetectionRules` array.

```js
assert.match(code, /const\s+checkpointDetectionRules\s*=\s*\[\s*player\.position\.x\s*>=\s*checkpoint\.position\.x,?\s*\]/)
```